chromium/rules.js

```diff
@@ -403,45 +403,62 @@ RuleSets.prototype = {
    */
   potentiallyApplicableRulesets: function(host) {
     // Have we cached this result? If so, return it!
-    var cached_item = this.ruleCache.get(host);
-    if (cached_item !== undefined) {
-      log(DBUG, "Ruleset cache hit for " + host + " items:" + cached_item.length);
+    if (this.ruleCache.has(host)) {
+      let cached_item = this.ruleCache.get(host);
+      let len = cached_item ? cached_item.size : 0;
+      log(DBUG, "Ruleset cache hit for " + host + " items:" + len);
       return cached_item;
-    }
-    log(DBUG, "Ruleset cache miss for " + host);
-
-    var tmp;
-    var results = [];
-    if (this.targets.has(host)) {
-      // Copy the host targets so we don't modify them.
-      results = results.concat(this.targets.get(host));
+    } else {
+      log(DBUG, "Ruleset cache miss for " + host);
     }
 
     // Ensure host is well-formed (RFC 1035)
-    if (host.indexOf("..") != -1 || host.length > 255) {
+    if (host.length > 255 || host.indexOf('..') != -1) {
       log(WARN,"Malformed host passed to potentiallyApplicableRulesets: " + host);
       return null;
     }
 
+    // Let's search!!
+    let resultSet = new Set();
+
+    // Look for exact matches
+    if (this.targets.has(host)) {
+      let results = this.targets.get(host);
+      if (results) {
```

Comment:
I think you can drop the `if (this.targets.has(host)) {` above since you do this `if (results) {` check here.

Reply:
ohhh wait, `results` can be `null` now hmm.